### PR TITLE
IBP-6063-adding_test_cases_to_add_and_remove_names_from_studies

### PR DESCRIPTION
**Pull request by cuenyad**
Please, review.

### CODE REVIEW

File: src/main/java/org/generationcp/middleware/api/germplasmlist/GermplasmListService.java [hunk 1]
```diff
@@ -98,4 +99,12 @@ void removeGermplasmEntriesFromList(Integer germplasmListId,
...
 	Map<Integer, Map<Integer, String>> getObservationValuesByListAndVariableIds(Integer listId, Set<Integer> variableIds);
 
+	/**
+	 * Inserts a list of multiple {@code GermplasmListData} objects into the database.
+	 *
+	 * @param data - A list of {@code GermplasmListData} objects to be persisted to the database. {@code GermplasmListData}
+	 *             objects must be valid.
+	 * @return Returns the ids of the {@code GermplasmListData} records inserted in the database.
+	 */
+	List<GermplasmListData> addGermplasmListData(List<GermplasmListData> data);
```
Comment (clarysabel) [inline src/main/java/org/generationcp/middleware/api/germplasmlist/GermplasmListService.java:109]: @cuenyad please do not expose this only to be used in a test. Call DAO directly using a simplified version of the implementation

File: src/test/java/org/generationcp/middleware/service/impl/study/StudyEntryServiceImplIntegrationTest.java [hunk 2]
```diff
@@ -406,7 +476,18 @@ false, new StudyType(6), "20200606", null, null,
 		this.daoFactory.getProjectPropertyDAO().save(entryNoProp);
 		this.daoFactory.getProjectPropertyDAO().save(crossProp);
 		this.daoFactory.getProjectPropertyDAO().save(entryTypeProp);
+		this.sessionProvder.getSession().refresh(plotDataset);
+		return study;
+	}
+
+	private DmsProject createStudyWithGermplasmListAsociated() {
```
Comment (clarysabel) [inline src/test/java/org/generationcp/middleware/service/impl/study/StudyEntryServiceImplIntegrationTest.java:483]: rename to createStudyWithGermplasmListAssociated

File: src/test/java/org/generationcp/middleware/service/impl/study/StudyEntryServiceImplIntegrationTest.java [hunk 3]
```diff
@@ -54,6 +61,7 @@ public class StudyEntryServiceImplIntegrationTest extends IntegrationTestBase {
 	private static final String ENTRYCODE = "ENTRYCODE-";
 	private static final String CROSS = "ABC/XYZ-";
 	private static final String SEEDSOURCE = "SEEDSOURCE-";
+	private static final String ACCNO = "ACCNO";
 
 	@Autowired
 	private StudyEntryService service;
```
Comment (clarysabel) [inline src/test/java/org/generationcp/middleware/service/impl/study/StudyEntryServiceImplIntegrationTest.java:67]: @cuenyad could you please rename service to studyEntryService ?

File: src/test/java/org/generationcp/middleware/service/impl/study/StudyEntryServiceImplIntegrationTest.java [hunk 4]
```diff
@@ -331,6 +348,59 @@ public void testGetStudyEntries() {
 		}
 	}
 
+	@Test
+	public void updatePlotDatasetProperties_shouldAddNameType() {
+		final DmsProject study = this.createStudyWithGermplasmListAsociated();
+
+		final List<StudyEntryColumnDTO> columns =
+			this.service.getStudyEntryColumns(study.getProjectId(), null);
+		final StudyEntryColumnDTO aCCNONameTypeColumn =
```
Comment (clarysabel) [inline src/test/java/org/generationcp/middleware/service/impl/study/StudyEntryServiceImplIntegrationTest.java:357]: rename to accNoNameTypeColumn

File: src/test/java/org/generationcp/middleware/service/impl/study/StudyEntryServiceImplIntegrationTest.java [hunk 5]
```diff
@@ -331,6 +348,59 @@ public void testGetStudyEntries() {
...
+	public void testUpdatePlotDatasetProperties_shouldRemoveNameType(){
+		final DmsProject study = this.createStudyWithGermplasmListAsociated();
+
+		final PlotDatasetPropertiesDTO plotDatasetPropertiesDTO = new PlotDatasetPropertiesDTO();
+		plotDatasetPropertiesDTO.setNameTypeIds(Arrays.asList(1));
+		plotDatasetPropertiesDTO.setVariableIds(Collections.emptyList());
+		this.studyDatasetService.updatePlotDatasetProperties(study.getProjectId(), plotDatasetPropertiesDTO, null);
+
+		final List<StudyEntryColumnDTO> columns = this.service.getStudyEntryColumns(study.getProjectId(), null);
+		StudyEntryColumnDTO aCCNONameTypeColumn =
```
Comment (clarysabel) [inline src/test/java/org/generationcp/middleware/service/impl/study/StudyEntryServiceImplIntegrationTest.java:388]: @cuenyad rename to accNoNameTypeColumn

File: src/main/java/org/generationcp/middleware/api/germplasmlist/GermplasmListService.java (hunk 1 above)
Comment (cuenyad) [reply src/main/java/org/generationcp/middleware/api/germplasmlist/GermplasmListService.java]: done

File: src/test/java/org/generationcp/middleware/service/impl/study/StudyEntryServiceImplIntegrationTest.java (hunk 2 above)
Comment (cuenyad) [reply src/test/java/org/generationcp/middleware/service/impl/study/StudyEntryServiceImplIntegrationTest.java]: done!

File: src/test/java/org/generationcp/middleware/service/impl/study/StudyEntryServiceImplIntegrationTest.java (hunk 3 above)
Comment (cuenyad) [reply src/test/java/org/generationcp/middleware/service/impl/study/StudyEntryServiceImplIntegrationTest.java]: done!

File: src/test/java/org/generationcp/middleware/service/impl/study/StudyEntryServiceImplIntegrationTest.java (hunk 4 above)
Comment (cuenyad) [reply src/test/java/org/generationcp/middleware/service/impl/study/StudyEntryServiceImplIntegrationTest.java]: done!

File: src/test/java/org/generationcp/middleware/service/impl/study/StudyEntryServiceImplIntegrationTest.java (hunk 5 above)
Comment (cuenyad) [reply src/test/java/org/generationcp/middleware/service/impl/study/StudyEntryServiceImplIntegrationTest.java]: done!

File: src/main/java/org/generationcp/middleware/api/germplasmlist/GermplasmListService.java [hunk 6]
```diff
@@ -7,6 +7,7 @@
 import org.generationcp.middleware.domain.inventory.common.SearchCompositeDto;
 import org.generationcp.middleware.domain.ontology.Variable;
 import org.generationcp.middleware.pojos.GermplasmList;
+import org.generationcp.middleware.pojos.GermplasmListData;
```
Comment (clarysabel) [inline src/main/java/org/generationcp/middleware/api/germplasmlist/GermplasmListService.java:10]: remove unused import @cuenyad

Comment (cuenyad) [reply src/main/java/org/generationcp/middleware/api/germplasmlist/GermplasmListService.java]: done!

**Pull request merged**